src/app/components/GridGallery/GridGallery.tsx

```diff
@@ -189,16 +123,62 @@ const GridGallery: React.FC = () => {
     document.body.classList.add('no-scroll');
     setOpenModal(index);
   }
+
```

Comment:
All  BL ( business logic ) should be moved out of compoennts. Components should always be dummy. They are not doing any job or have any side effects.